src/backend/PostgresBackend.cpp

```diff
@@ -393,153 +450,99 @@ PostgresBackend::doFetchSuccessorKey(
 
 std::vector<TransactionAndMetadata>
 PostgresBackend::fetchTransactions(
-    std::vector<ripple::uint256> const& hashes) const
+    std::vector<ripple::uint256> const& hashes,
+    boost::asio::yield_context& yield) const
 {
+    if (!hashes.size())
+        return {};
+
     std::vector<TransactionAndMetadata> results;
-    constexpr bool doAsync = true;
-    if (doAsync)
-    {
-        auto start = std::chrono::system_clock::now();
-        auto end = std::chrono::system_clock::now();
-        auto duration = ((end - start).count()) / 1000000000.0;
-        results.resize(hashes.size());
-        std::condition_variable cv;
-        std::mutex mtx;
-        std::atomic_uint numRemaining = hashes.size();
-        for (size_t i = 0; i < hashes.size(); ++i)
-        {
-            auto const& hash = hashes[i];
-            boost::asio::post(
-                pool_, [this, &hash, &results, &numRemaining, &cv, &mtx, i]() {
-                    BOOST_LOG_TRIVIAL(debug)
-                        << __func__ << " getting txn = " << i;
-                    PgQuery pgQuery(pgPool_);
-                    std::stringstream sql;
-                    sql << "SELECT transaction,metadata,ledger_seq,date FROM "
-                           "transactions "
-                           "WHERE HASH = \'\\x"
-                        << ripple::strHex(hash) << "\'";
-
-                    auto res = pgQuery(sql.str().data());
-                    if (size_t numRows = checkResult(res, 4))
-                    {
-                        results[i] = {
-                            res.asUnHexedBlob(0, 0),
-                            res.asUnHexedBlob(0, 1),
-                            res.asBigInt(0, 2),
-                            res.asBigInt(0, 3)};
-                    }
-                    if (--numRemaining == 0)
-                    {
-                        std::unique_lock lck(mtx);
-                        cv.notify_one();
-                    }
-                });
-        }
-        std::unique_lock lck(mtx);
-        cv.wait(lck, [&numRemaining]() { return numRemaining == 0; });
-        auto end2 = std::chrono::system_clock::now();
-        duration = ((end2 - end).count()) / 1000000000.0;
-        BOOST_LOG_TRIVIAL(info)
-            << __func__ << " fetched " << std::to_string(hashes.size())
-            << " transactions with threadpool. took "
-            << std::to_string(duration);
+    PgQuery pgQuery(pgPool_);
+    pgQuery("SET statement_timeout TO 10000", yield);
+    std::stringstream sql;
+    for (size_t i = 0; i < hashes.size(); ++i)
+    {
+        auto const& hash = hashes[i];
+        sql << "SELECT transaction,metadata,ledger_seq,date FROM "
+               "transactions "
+               "WHERE HASH = \'\\x"
+            << ripple::strHex(hash) << "\'";
+        if (i + 1 < hashes.size())
+            sql << " UNION ALL ";
     }
-    else
+    auto start = std::chrono::system_clock::now();
```

Comment:
This may exceed the scope of this pull request, and I'm fine with that. However, there are several places in the code where we time the execution of various functions. Consider using [`boost::cpu_timer`](https://www.boost.org/doc/libs/1_75_0/libs/timer/doc/cpu_timers.html) to avoid manually calculating the duration.